benchmarks/src/commonMain/kotlin/ReadStringBenchmark.kt

```diff
@@ -0,0 +1,42 @@
+package kotlinx.io.benchmarks
+
+import kotlinx.benchmark.Benchmark
+import kotlinx.benchmark.Param
+import kotlinx.benchmark.Scope
+import kotlinx.benchmark.Setup
+import kotlinx.benchmark.State
+import kotlinx.io.Buffer
+import kotlinx.io.Source
+import kotlinx.io.readCodePointValue
+import kotlinx.io.readString
+import kotlinx.io.writeCodePointValue
+import kotlinx.io.writeString
+import kotlin.random.Random
+
+
+@State(Scope.Benchmark)
+open class ReadStringBenchmark() {
+
+    @Param("16", "64", "512")
+    var size: Int = 0
```

Comment:
Did you intentionally pick size values so that a string always fit into a single segment? If so, could you please drop a comment in the code.

It would be also nice to mention why we have this benchmark along with `Utf8StringBenchmark`.

Reply:
Yes, dropped a comment.

>It would be also nice to mention why we have this benchmark along with Utf8StringBenchmark.

My general approach is the following -- there are "generic" benchmarks that are the overall proxy to the performance of a particular method/pipeline. E.g. full-blown JSON transformation, various modes of UTF decoding etc. They are good to measure the impact of the significant change.

On the other hand, there are "tailored" ones that are written to specifically pinpoint some code-pattern or behavior.

`Utf8StringBenchmark` is the former, this one is the latter. I do not mind changing it as you see fit